src/main/java/redis/embedded/RealRedisCluster.java

```diff
@@ -0,0 +1,281 @@
+package redis.embedded;
+
+import redis.clients.jedis.Jedis;
+import redis.embedded.exceptions.EmbeddedRedisException;
+
+import java.util.*;
+
+/**
+ * Created by dragan on 17.07.15.
+ */
+public class RealRedisCluster implements Cluster {
+    private static final int CLUSTER_HASH_SLOTS_NUMBER = 16384;
+    private static final String LOCAL_HOST = "127.0.0.1";
+
+    private final List<Redis> servers = new LinkedList<Redis>();
+    private final int numOfReplicates;
+
+    RealRedisCluster(List<Redis> servers, int numOfReplicates) {
+        validateParams(servers, numOfReplicates);
+        this.servers.addAll(servers);
+        this.numOfReplicates = numOfReplicates;
+    }
+
+    RealRedisCluster(List<Redis> servers) {
+        validateParams(servers, 1);
+        this.servers.addAll(servers);
+        this.numOfReplicates = 1;
+    }
+
+    private void validateParams(List<Redis> servers, int numOfReplicates) {
+        if (servers.size() <= 2) {
+            throw new EmbeddedRedisException("Redis Cluster requires at least 3 master nodes.");
+        }
+        if (numOfReplicates < 1) {
+            throw new EmbeddedRedisException("Redis Cluster requires at least 1 replication.");
+        }
+        if (numOfReplicates > servers.size() - 1) {
+            throw new EmbeddedRedisException("Redis Cluster requires number of replications less than number of nodes - 1.");
+        }
+    }
+
+    @Override
+    public void create() throws EmbeddedRedisException {
```

Comment:
Do we really need separate create() or can we call create() as a part of start()?